sketches/src/main/java/org/apache/apex/malhar/sketches/QuantilesEstimator.java

```diff
@@ -0,0 +1,188 @@
+{
+  private transient QuantilesSketch quantilesSketch = QuantilesSketch.builder().build();
+
+  /**
+   * Constructor that allows non-default initialization of the quantile sketch object
+   *
+   * @param k:    Parameter that determines accuracy and memory usage of quantile sketch. See QuantilesSketch documentation
+   *              for details
+   * @param seed: The quantile sketch algorithm is inherently random. Set seed to 0 for reproducibility in testing, but
+   *              do not set otherwise.
+   */
+  public QuantilesEstimator(int k, short seed)
+  {
+    quantilesSketch = QuantilesSketch.builder().setSeed(seed).build(k);
+  }
+
+  /**
+   * Output port that emits cdf estimated at the current data point
+   */
+  public final transient DefaultOutputPort<Double> cdfOutput = new DefaultOutputPort<>();
+  /**
+   * Emits quantiles of stream seen thus far
+   */
+  public final transient DefaultOutputPort<double[]> quantilesOutput = new DefaultOutputPort<>();
+  /**
+   * Emits probability masses on specified intervals
+   */
+  public final transient DefaultOutputPort<double[]> pmfOutput = new DefaultOutputPort<>();
+  /**
+   * This operator computes three different quantities which are output on separate output ports. If not using any of
+   * these quantities, these variables can be set to avoid unnecessary computation.
+   */
+
+  private boolean computeCdf = true;
+  private boolean computeQuantiles = true;
+  private boolean computePmf = true;
+
+  public boolean isComputeCdf()
+  {
+    return computeCdf;
+  }
+
+  public void setComputeCdf(boolean computeCdf)
+  {
+    this.computeCdf = computeCdf;
+  }
+
+  public boolean isComputeQuantiles()
+  {
+    return computeQuantiles;
+  }
+
+  public void setComputeQuantiles(boolean computeQuantiles)
+  {
+    this.computeQuantiles = computeQuantiles;
+  }
+
+  public boolean isComputePmf()
+  {
+    return computePmf;
+  }
+
+  public void setComputePmf(boolean computePmf)
+  {
+    this.computePmf = computePmf;
+  }
+
+  public int getK()
+  {
+    return quantilesSketch.getK();
+  }
+
+  public double[] getFractions()
+  {
+    return fractions;
+  }
+
+  public void setFractions(double[] fractions)
+  {
+    this.fractions = fractions;
+  }
+
+  public double[] getPmfIntervals()
+  {
+    return pmfIntervals;
+  }
+
+  public void setPmfIntervals(double[] pmfIntervals)
+  {
+    this.pmfIntervals = pmfIntervals;
+  }
+
+  /**
+   * Returns total number of elements seen in stream so far.
+   */
+  public long streamLength() {
+    return quantilesSketch.getN();
+  }
+
+  /**
+   * This field determines the specific quantiles to be calculated.
+   * Default is set to compute the standard quartiles.
+   */
+  private double[] fractions = {0.0, 0.25, 0.50, 0.75, 1.00};
+  /**
+   * This field determines the intervals on which the probability mass function is computed.
+   */
+  private double[] pmfIntervals = {};
+
+  /**
+   * Input port takes a number
```

Comment:
This comment is not helpful. Input port also does not take a number, it takes a double.